lib/mediawiki/query/lists/search.rb

```diff
@@ -36,6 +36,24 @@ def get_search_results(search_value, namespace = 0)
           query_ary(params, 'search', 'title')
         end
 
+        # Gets an array containing page titles that matched the search.
+        # @param search_value [String] The thing to search for.
+        # @param namespace [Fixnum] The namespace to search in. Defaults to 0 (the main namespace).
+        # @see https://www.mediawiki.org/wiki/API:Search MediaWiki Search API Docs
+        # @since 3.0.1
+        # @return [Array<String>] The page titles that matched the search.
+        def get_search_text_results(search_value, namespace = 0)
+          params = {
+            list: 'search',
+            srsearch: search_value,
+            srwhat: 'text',
+            srlimit: '500',
```

Comment:
The limit should not be hardcoded. The current standard is to have a `limit` parameter which defaults to `@query_limit_default`, and then call `get_limited` on that value. As in:
```ruby
def get_search_text_results(search_value, namespace = 0, limit = @query_limit_default)
...
    srlimit: get_limited(limit)
...
```